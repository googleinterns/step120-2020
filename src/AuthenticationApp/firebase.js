import firebase from "firebase/app";
import "firebase/auth";
import "firebase/firestore";
import "firebase/analytics";
import APIKey from "./APIKey";
 
const firebaseConfig = {
    apiKey: APIKey,
    authDomain: "step120-2020.firebaseapp.com",
    databaseURL: "https://step120-2020.firebaseio.com",
    projectId: "step120-2020",
    storageBucket: "step120-2020.appspot.com",
    messagingSenderId: "994372108603",
    appId: "1:994372108603:web:cb5d95fa2ec4d107b1b71e",
    measurementId: "G-C0ZHMWXRWH"
  };
 
  firebase.initializeApp(firebaseConfig);
  //firebase.analytics();
  export const generateUserDocument = async (user, furtherData) => {
      if(!user) return;
      
      //fill doc with user data
      
      return getUserDocument(user.uid);
  };
  const getUserDocument = async uid => {
      if (!uid) return null;
      try {
          const userDocument = await firestore.doc(`users/${uid}`).get();
          return {
              uid,
              ...userDocument.data()
          };
      } catch (error) {
          console.error("Error fetching user", error);
      }
  };
  export const authentication = firebase.auth();
  export const firestore = firebase.firestore();
