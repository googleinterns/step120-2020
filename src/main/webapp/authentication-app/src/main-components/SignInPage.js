import React, {useState} from "react";
import {Link} from "@reach/router";

const SignInPage = () => {
  const [userPassword, setUserPassword] = useState('');
  const [userEmail, setUserEmail] = useState('');
  const [error, setError] = useState(null);

  const signInHandler = (event,email,password) => {
    event.preventDefault();
  };

  const onChangeHandler = (event) => {
      const {name, value} = event.currentTarget;
      {/* gets the . part of the address to check if its edu */}
      const emailEnd = name.substring(name.length - 4, name.length);
      
      {/*
        TODO redirect the user to an error page
        that tells them that their email addresses
        is invalid and must be a .edu address 
        */}
      if(name === 'userEmail') {
        setUserEmail(value);
      }
      else if(name === 'userPassword') {
        setUserPassword(value);
      }
  };

  return (
      <div className="SignIn">
        <h1 className="Title">Sign In </h1>
        <div className="SignIn-Area">
          <form className="SignIn-form">
            <label htmlFor="userEmail" className="block">
              Your Email:
            </label>
            <input
              type="userEmail"
              className=""
              name="userEmail"
              value= {userEmail}
              placeholder= "E.x: JohnAppleseed@gmail.com"
              id="userEmail"
              onChange = {(event) => onChangeHandler(event)} 
            />
            <label htmlFor="userPassword" className="block">
              Make a Strong Password:
            </label>
            <input
              type="userPassword"
              className=""
              name="userPassword"
              value= {userPassword}
              placeholder="Your Password Here"
              onChange = {(event) => onChangeHandler(event)}
            />
            <button 
              className="SignInButton"
              onClick= {(event) => {signInHandler(event, userEmail, userPassword)}}>
              Click to Sign in
            </button>
          </form>
          <p className="">or</p>
          <button> Sign in with Google </button>
          <p className="">
            Don't have an account {" "}
            <Link to = "signUp" className="signup-link">
              Sign up 
            </Link>{" "}
            <br /> {" "}
            <Link to = "passwordReset" className="password-reset-link">
              Forgot your password? Reset it here
            </Link>
          </p>
        </div>
      </div>

        
  );
}
export default SignInPage;
