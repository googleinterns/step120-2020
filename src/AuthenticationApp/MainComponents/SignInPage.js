import React, {useState} from "react";
import {Link, navigate} from "@reach/router";
import { authentication } from "../firebase";
 
const SignInPage = () => {
  const [userPassword, setUserPassword] = useState('');
  const [userEmail, setUserEmail] = useState('');
  const [error, setError] = useState(null);
 
  const signInHandler = async (event,email,password) => {
    event.preventDefault();
    authentication.signInWithEmailAndPassword(email, password).catch(error => {
        setError("Error signing in with password and email!");
        console.error("Error signing in with password and email", error);
    });
    if (authentication.currentUser) {
        navigate("/");
    }
  };
  
  const onUserPasswordChangeHandler = event => {
        const {value} = event.currentTarget;
        setUserPassword(value);
    }
    const onUserEmailChangeHandler = event => {
        const {value} = event.currentTarget;
        setUserEmail(value);
    }
 
  return (
      <div className="SignIn">
        <h1 className="Title">Sign In </h1>
        <div className="SignIn-Area">
          {error !== null && <div>{error}</div>}
          <form className="SignIn-form">
            <label htmlFor="userEmail" className="block">
              Your Email:
            </label>
            <input
              type="userEmail"
              className=""
              name="userEmail"
              value= {userEmail}
              placeholder= "E.x: JohnAppleseed@gmail.com"
              id="userEmail"
              onChange = {(event) => onUserEmailChangeHandler(event)} 
            />
            <label htmlFor="userPassword" className="block">
              Your Password:
            </label>
            <input
              type="password"
              className=""
              name="userPassword"
              value= {userPassword}
              placeholder="Your Password Here"
              onChange = {(event) => onUserPasswordChangeHandler(event)}
            />
            <button 
              className="SignInButton"
              onClick= {(event) => {signInHandler(event, userEmail, userPassword)}}>
              Click to Sign in
            </button>
          </form>
          <p className="">or</p>
          <button> Sign in with Google </button>
          <p className="">
            Don't have an account {" "}
            <Link to = "signUp" className="signup-link">
              Sign up 
            </Link>{" "}
            <br /> {" "}
            <Link to = "passwordReset" className="password-reset-link">
              Forgot your password? Reset it here
            </Link>
          </p>
        </div>
      </div>
  );
}
export default SignInPage;
