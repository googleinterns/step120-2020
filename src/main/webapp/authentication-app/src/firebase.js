import firebase from "firebase/app";
import "firebase/auth";
import "firebase/firestore";

const firebaseConfig = {
    apiKey: "AIzaSyAjse2YjCJHUUe5w1i7I9jkjH8cqqRcAIU",
    authDomain: "step120-2020.firebaseapp.com",
    databaseURL: "https://step120-2020.firebaseio.com",
    projectId: "step120-2020",
    storageBucket: "step120-2020.appspot.com",
    messagingSenderId: "994372108603",
    appId: "1:994372108603:web:cb5d95fa2ec4d107b1b71e",
    measurementId: "G-C0ZHMWXRWH"
  };

  firebase.initializeApp(firebaseConfig);
  firebase.analytics();
  export const authentication = firebase.auth();
  export const firestore = firebase.firestore();