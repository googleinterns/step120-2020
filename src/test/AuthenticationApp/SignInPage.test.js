import * as firebase from 'firebase';
import React from 'react';
import ReactDOM from 'react-dom';
import SignInPage from '../../AuthenticationApp/MainComponents/SignInPage';
import { shallow } from "enzyme";
import { mount } from "enzyme";
import { create } from "react-test-renderer";
import renderer from 'react-test-renderer'

import { configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
configure({ adapter: new Adapter() });


describe('<SignInPage />', () => {
    let wrapper;
    const setState = jest.fn();
    const useStateSpy = jest.spyOn(React, 'useState');
    useStateSpy.mockImplementation((init) => [init, setState]);

    afterEach(() => {
        jest.clearAllMocks();
    });

    describe("Snapshot Test" , () => {
        it("should render without crashing", () => {
            const domDiv = document.createElement('div');
            ReactDOM.render(<SignInPage />, domDiv);
        });

        it("renders a snapshot of Component", () => {
            const snapshot = renderer.create(<SignInPage />).toJSON();
            expect(snapshot).toMatchSnapshot();
        })

    })

    it("should update userPassword state", () => {
        wrapper = mount(<SignInPage onUserPasswordChangeHandler={setState} />);
        
        wrapper.find('.userPassword').simulate('change');
        expect(setState).toBeTruthy();
    });

    it("should update userEmail state", () => {
        wrapper = mount(<SignInPage onUserEmailChangeHandler={setState} />);
        
        wrapper.find('.userEmail').simulate('change');
        expect(setState).toBeTruthy();
    });
    
});