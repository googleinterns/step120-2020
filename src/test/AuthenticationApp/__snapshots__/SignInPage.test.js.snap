// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SignInPage /> Snapshot Test renders a snapshot of Component 1`] = `
<div
  className="SignIn"
>
  <h1
    className="Title"
  >
    Sign In 
  </h1>
  <div
    className="SignIn-Area"
  >
    <form
      className="SignIn-form"
    >
      <label
        className="block"
        htmlFor="userEmail"
      >
        Your Email:
      </label>
      <input
        className="userEmail"
        id="userEmail"
        name="userEmail"
        onChange={[Function]}
        placeholder="E.x: JohnAppleseed@gmail.com"
        type="userEmail"
        value=""
      />
      <label
        className="block"
        htmlFor="userPassword"
      >
        Your Password:
      </label>
      <input
        className="userPassword"
        name="userPassword"
        onChange={[Function]}
        placeholder="Your Password Here"
        type="password"
        value=""
      />
      <button
        className="signInButton"
        onClick={[Function]}
      >
        Click to Sign in
      </button>
    </form>
    <p
      className=""
    >
      or
    </p>
    <button>
       Sign in with Google 
    </button>
    <p
      className=""
    >
      Don't have an account 
       
      <a
        className="signup-link"
        href="/signUp"
        onClick={[Function]}
      >
        Sign up
      </a>
       
      <br />
       
       
      <p>
        Forgot your password? 
         
        <a
          className="password-reset-link"
          href="/passwordReset"
          onClick={[Function]}
        >
          Reset 
           
        </a>
        it here!
      </p>
    </p>
  </div>
</div>
`;
